lib/streamlit/dataframe_util.py

```diff
@@ -805,9 +885,9 @@ def convert_anything_to_arrow_bytes(
     if isinstance(data, pa.Table):
         return convert_arrow_table_to_arrow_bytes(data)
 
-    if is_pandas_data_object(data):
-        # All pandas data objects should be handled via our pandas
-        # conversion logic. We are already calling it here
+    if is_pandas_data_object(data) and is_unevaluated_data_object(data):
+        # All pandas and unevaluated data objects should be handled via
```

Comment:
this comment sounds like it would be an `or` check. I would rephrase it to `All unevaluated pandas data objects ...`

Reply:
oh, good catch. It needs to be `or`

Reply:
Ahh ok great 🙂 In case this was not caught by a test, now would be a good time to add a test case for this 😁

Reply:
yep, I will add a test 👍

Reply:
Thought about it a bit more. I'm not yet feeling fully confident that we can correctly handle more data objects directly via raw arrow. So, I cleaned this up for now to almost always use the Pandas fallback. I need to investigate it a bit more and will put all the raw Arrow handling into a dedicated PR with better testing (maybe after the next release).